metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/OrchestratorService.java

```diff
@@ -539,12 +540,23 @@ public DatasetExecutionInformation getDatasetExecutionInformation(MetisUser meti
     AbstractMetisPlugin firstPublishPlugin = workflowExecutionDao
         .getFirstFinishedWorkflowExecutionPluginByDatasetIdAndPluginType(datasetId, EnumSet
             .of(PluginType.PUBLISH));
-    AbstractMetisPlugin lastPublishPlugin = workflowExecutionDao
-        .getLastFinishedWorkflowExecutionPluginByDatasetIdAndPluginType(datasetId, EnumSet
-            .of(PluginType.PUBLISH));
     AbstractMetisPlugin lastPreviewPlugin = workflowExecutionDao
         .getLastFinishedWorkflowExecutionPluginByDatasetIdAndPluginType(datasetId, EnumSet
             .of(PluginType.PREVIEW));
+    AbstractMetisPlugin lastPublishPlugin = workflowExecutionDao
+        .getLastFinishedWorkflowExecutionPluginByDatasetIdAndPluginType(datasetId, EnumSet
+            .of(PluginType.PUBLISH));
+
+    final WorkflowExecution runningOrInQueueExecution = workflowExecutionDao
+        .getRunningOrInQueueExecution(datasetId);
```

Comment:
What if runningOrInQueueExecution is null? I don't think we can guarantee that we currently have an execution?